NumLib/NamedFunctionCaller.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_NAMEDFUNCTIONCALLER_H
+#define NUMLIB_NAMEDFUNCTIONCALLER_H
+
+#include <map>
+#include <vector>
+#include "NamedFunction.h"
+
+namespace NumLib
+{
+class SpecialFunctionCaller;
+
+//! Builds expression trees of named functions dynamically at runtime.
+class NamedFunctionCaller final
+{
+public:
+    //! Constructs an instance whose unbound arguments have the given names.
+    explicit NamedFunctionCaller(
+        std::initializer_list<std::string> unbound_arguments);
+
+    //! Adds the given named function
+    void addNamedFunction(NamedFunction&& fct);
+    std::vector<NamedFunction> const& getNamedFunctions() const
+    {
+        return _named_functions;
+    }
+
+    //! Declares that the argument with name \c sink_args of the function \c
+    //! sink_fct is being computed by the function \c source.
+    //!
+    //! The functions involved need not already be known to the
+    //! NamedFunctionCaller.
+    void plug(std::string const& sink_fct, std::string const& sink_arg,
```

Comment:
That's a bit related to your question above.
`plug()` defines edges in the expression call graph. Upon calling plug the associated vertices (i.e. named functions) needn't be there yet (That is, because when parsing the process configuration not all of the inner parts of the process exist yet.).
However, upon calling `applyPlugs()` all needed named function must have been added to the `NamedFunctionCaller`.
So: No I think it's not possible easily.